File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch02/Python/P02-Fact_comb.py
# Calculates combinations using a factorial function

#============================================================================
def Fact(n):
#----------------------------------------------------------------------------
#  Returns the factorial of n
#----------------------------------------------------------------------------
   f = 1e0
   for i in range(2,n+1): f *= i
   return f

# main

n = int(input("n = "))
k = int(input("k = "))

C = Fact(n)/(Fact(k)*Fact(n-k))

print("C({0:d},{1:d}) = {2:f}".format(n,k,C))

File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch02/C/P02-Fact_comb.cpp
// Calculates combinations using a factorial function
#include <stdio.h>

//===========================================================================
double Fact(int n)
//---------------------------------------------------------------------------
// Returns the factorial of n
//---------------------------------------------------------------------------
{
   double f;
   int i;

   f = 1e0;
   for (i=2; i<=n; i++) f *= i;
   return f;
}

int main()
{
   double C;
   int k, n;

   printf("n = "); scanf("%d",&n);
   printf("k = "); scanf("%d",&k);

   C = Fact(n)/(Fact(k)*Fact(n-k));

   printf("C(%d,%d) = %f\n",n,k,C);
}
